src/controller/java/src/chip/devicecontroller/ControllerParams.java

```diff
@@ -248,8 +302,34 @@ public Builder setIpk(byte[] ipk) {
     }
 
     /**
-     * Sets the AdminSubject value passed to ChipDeviceCommissioner's CommissioningParameters. This
-     * value is passed in the AddNoc command sent to the commissionee and represents the subject of
+     * The Product Attestation Authority certificates that are trusted to sign
+     * device attestation information.
+     * 
+     * @param paaCerts The Product Attestation Authority certificates
+     *                 containing the X.509 DER certificate.
+     */
+    public Builder setPaaCerts(ArrayList<byte[]> paaCerts) {
+      this.paaCerts = paaCerts;
+      return this;
+    }
+
+    /**
+     * The Certificate Declaration certificates that are trusted to sign
+     * device attestation information.
+     * 
+     * @param cdCerts The Certificate Declaration certificates
+     *                containing the X.509 DER certificate.
+     */
+    public Builder setCdCerts(ArrayList<byte[]> cdCerts) {
```

Comment:
Instead of allow the app to overwrite the trustStore, it seems your API is to only allow the app to overwrite the cdCerts, right?

Reply:
`setCerts` does not belong to trustStore, it is a part of deviceAttestaionVerifier. I didn't want to implement this api, but when I saw Darwin implement this function, I did it by the way.